### DISCUSSION TITLE
Define the 0.2 demo

### DISCUSSION THREAD

**Discussion by gbrivady** [iScsc/wiresmash#11] [Ideas]
Discussion thread for #9

**Comment by gbrivady** [comment]
We will probably discuss it on the briefing call, but feel free to throw ideas here so we can already think about them.

The demo should not be too ambitious, but mostly aim to test if we can easily add new subsystems to the current demo. So I propose we add the following systems:

- Working combat demo: need to be able to inflict and receive damage, have an health bar
- Ehances physic subsystem: make a real newton law, add a max speed
- Working jump feature : instead of the current awkward teleportation, add a proper jump with speed increase, lock out of movements while jumping, etc...

But I feel like the third point could on its own be another demo, but we could probably work on it as soon as point 2 is ready.

**Reply by ctmbl** [reply]
I don't have any idea of the current code state but 
> Working combat demo: need to be able to inflict and receive damage, have an health bar

seems quite ambitious, maybe I'm overestimating the feature though
However you're the most experienced one so I trust you to make the right decision!

**Reply by gbrivady** [reply]
> I don't have any idea of the current code state but

I think I am the only one who know until we do the planned meeting and/or people start seriously tackling the PR to be fair 😅


> seems quite ambitious, maybe I'm overestimating the feature though

Most of the work is already there, you are right in trusting me 👀 

But that is the whole point of the discussion, thinking about the "doablety" of everyone's idea, so if you want me to give more details on why I think it is easily doable, just ask!

**Reply by ctmbl** [reply]
> I think I am the only one who know until we do the planned meeting and/or people start seriously tackling the PR to be fair sweat_smile

absolutely ahah

> But that is the whole point of the discussion, thinking about the "doablety" of everyone's idea, so if you want me to give more details on why I think it is easily doable, just ask!

Overall if you're certain about the working combat demo I quite like your Milestone! 
it's a go for me, you can create it and update it if anyone has something to add
